### DISCUSSION TITLE
Exporting Expressions for All Containers

### DISCUSSION THREAD

**Discussion by monkeylove** [Cookie-AutoDelete/Cookie-AutoDelete#1060] [Q&A / Support] [answered]
Hello. When I export expressions, is it done only for the selected container or all containers?

I'm not sure, but I think when I import, I only get the rules for the default container. The rest of the containers are empty.

Thank you for your help on the matter.

**Comment by kennethtran93** [comment]
Exporting expressions export all lists and their expressions. If you have created custom containers, then you'll have to figure out the ID of the new container in the browser that you want to import, with the exception of the default containers populated by Firefox. We do not create any new containers, so any custom containers created may show up as invalid container.

Dont forget to enable containers tabs on the settings again to see the default container lists. The settings export is kept separately for now.

**Comment by monkeylove** [comment]
Thank you. To clarify, I also used Multi-Account Containers. I have, for example, two containers: Default (no container), and monkeylove.

In CookieAutoDelete, enable support for container tabs is enabled.

In the list of expressions, I have two tabs: default (no container) and firefox-container-24 (monkeylove).

When I export expressions, then the lists of both containers will be exported.

If, for some reason, I have to reinstall the addon or reinstall all addons, do I do it this way?

First, install Multi-Account Containers, then restore the containers via sync.

Then, install Cookie Autodelete and make sure that enable support for container tabs is enabled.

Third, import the expressions.

The lists for default and the monkeylove container should be imported correctly.

**Comment by kennethtran93** [comment, marked as answer]
The syncing of containers is tricky. I do not know how container sync happens on Firefox (whether or not they'll keep the original container id of your other browser). 

I suggest that once you have your containers synced over and this extension installed, manually create a dummy expression (with the name of the container for easier finding) on the list where your expressions should go, then export that configuration.
Open that exported file in a text editor, search for that dummy expression and see if the container id matches. If so, you can remove your dummy entry and just import the old list as is.

If it is different, then you will need to manually edit each expression to reflect the new container ID of the dummy expression, as well as the container ID of the list itself. There is documentation on how the data should be formatted for import on the wiki side.

I have yet to make this process easier on the backend, so it is all a manual process for now.
